--- libjingle-0.6.14/configure.ac~	2012-07-07 09:55:02.000000000 +0300
+++ libjingle-0.6.14/configure.ac	2012-07-07 09:55:28.960709537 +0300
@@ -48,8 +48,8 @@
     AC_SUBST(OPENSSL_LIBS)
     AC_DEFINE(SSL_USE_OPENSSL, 1, [Using OpenSSL])
     AC_DEFINE(FEATURE_ENABLE_SSL, 1, [SSL Enabled])
-    CPPFLAGS+="$OPENSSL_CFLAGS"
-    LDFLAGS+=" $OPENSSL_LIBS"
+    CPPFLAGS="$CPPFLAGS "$OPENSSL_CFLAGS""
+    LDFLAGS="$LDFLAGS "$OPENSSL_LIBS""
   else
     AC_DEFINE(HAVE_OPENSSL_SSL_H, 0, [Did not find openssl/ssl.h])
   fi
